onadata/apps/logger/management/commands/populate_submission_counters.py

```diff
@@ -0,0 +1,190 @@
+                    date=Cast(
+                        Concat(
+                            F('year'), Value('-'), F('month'), Value('-'), 1
+                        ),
+                        DateField(),
+                    )
+                ).filter(user_id=user.pk, date__gte=date_threshold).delete()
+
+                for xf in user.xforms.only('pk').iterator(chunk_size=chunks):
+                    daily_counters = []
+                    for values in (
+                        xf.instances.filter(
+                            date_created__date__gte=date_threshold
+                        )
+                        .values('date_created__date')
+                        .annotate(num_of_submissions=Count('pk'))
+                        .order_by('date_created__date')
+                    ):
+                        submission_date = values['date_created__date']
+                        daily_counters.append(DailyXFormSubmissionCounter(
+                            xform_id=xf.pk,
+                            date=submission_date,
+                            counter=values['num_of_submissions'],
+                        ))
+                        key = (
+                            f'{submission_date.year}-{submission_date.month}'
+                        )
+                        total_submissions[key] += values['num_of_submissions']
+
+                    if daily_counters:
+                        if verbosity >= 2:
+                            self.stdout.write(f'\tInserting daily counters data...')
+                        DailyXFormSubmissionCounter.objects.bulk_create(
+                            daily_counters, batch_size=chunks
+                        )
+                    elif verbosity >= 2:
+                        self.stdout.write(f'\tNo daily counters data...')
+
+                for key, total in total_submissions.items():
+                    year, month = key.split('-')
+                    monthly_counters.append(MonthlyXFormSubmissionCounter(
+                        year=year,
+                        month=month,
+                        xform_id=xf.pk,
+                        user_id=user.pk,
+                        counter=total,
+                    ))
+
+                if monthly_counters:
+                    if verbosity >= 2:
+                        self.stdout.write(f'\tInserting monthly counters data...')
+                    MonthlyXFormSubmissionCounter.objects.bulk_create(
+                        monthly_counters, batch_size=chunks
+                    )
+                elif verbosity >= 2:
+                    self.stdout.write(f'\tNo monthly counters data!')
+
+                # Update user's profile (and lock the related row)
+                updates = {
+                    'submissions_suspended': False,
+                    'counters_updates_status': 'complete',
+                }
+                UserProfile.objects.select_for_update().filter(
```

Comment:
I think `select_for_update` isn't needed since the `UPDATE` runs as an atomic operation within Postgres